linter_api/src/ast/item/use_decl_item.rs

```diff
@@ -30,6 +30,17 @@ pub struct UseDeclItem<'ast> {
 
 super::impl_item_data!(UseDeclItem, UseDecl);
 
+#[repr(C)]
+#[non_exhaustive]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum UseKind {
```

Comment:
Update `UseItem` to be like described in the reference and not how rustc handles it. This will also remove this enum